itou/www/apply/views/submit_views.py

```diff
@@ -172,6 +174,49 @@ def step_create_job_seeker(
     return render(request, template_name, context)
 
 
+@login_required
+@valid_session_required
+def step_check_prev_applications(
+    request, siae_pk, template_name="apply/submit_step_check_prev_applications.html"
+):
+    """
+    Check previous job applications to avoid duplicates.
+    """
+    session_data = request.session[settings.ITOU_SESSION_JOB_APPLICATION_KEY]
+    siae = get_object_or_404(Siae.active_objects, pk=session_data["to_siae_pk"])
+    job_seeker = get_user_model().objects.get(pk=session_data["job_seeker_pk"])
+
+    prev_applications = job_seeker.job_applications.filter(to_siae=siae)
+
+    # Limit the possibility of applying to the same SIAE for 24 hours.
+    if prev_applications.created_in_past_hours(24).exists():
+        raise PermissionDenied(
+            _(
+                "Vous avez déjà postulé chez cet employeur durant les dernières 24 heures."
+            )
+        )
+
+    next_url = reverse("apply:step_eligibility", kwargs={"siae_pk": siae.pk})
+
+    if not prev_applications.exists():
+        return HttpResponseRedirect(next_url)
+
+    # If a candidate is applying to an SIAE where he or she has already applied,
+    # ask for a confirmation to apply again.
```

Comment:
```
# At this point we know that the candidate is applying to an SIAE
# where he or she has already applied.

# Allow a new application if the user confirmed it despite the duplication warning.
```

Reply:
Bikeshedding spotted! 🤣

Reply:
TIL bikeshedding :-) _The term comes from an illustrative anecdote of a committee discussing a plan to build a nuclear power plant. In their meeting they spend the majority of their time arging over the color to paint the bikeshed in the back, because that was the part of the plan that everybody could understand._

En fait ton commentaire original m'a confusé parce que juste au dessous de lui le code ne correspondait pas... :D Mais bon oui tu peux skipper hihi

Reply:
bike shed
n abri (m) à bicyclettes

Reply:
Ah, ça correspondait pour moi, mais c'est bien d'avoir un regard extérieur, je prends le feedback.